File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    
    var answer = [0,0]
    while(s!=='1'){
        s=s.split("")
        let temp = s.filter(v =>v==='1').length
        answer[0]++;
        answer[1] +=s.length-temp
        s = temp.toString(2)    
    }
     return answer
    
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[]{0, 0};
    while (!s.equals("1")) {
        char[] sArr = s.toCharArray();
        int temp = 0;
        for (char c : sArr) {
            if (c == '1') {
                temp++;
            }
        }
        answer[0]++;
        answer[1] += sArr.length - temp;
        s = Integer.toBinaryString(temp);
    }
    return answer;
    }
}
